python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2

```diff
@@ -1268,25 +1278,128 @@ router bgp {{ router_bgp.as }}
+{%                 set redistribute_host = "redistribute attached-host" %}
+{%                 if redistribute_var.attached_host.route_map is arista.avd.defined %}
+{%                     set redistribute_host = redistribute_host ~ " route-map " ~ redistribute_var.attached_host.route_map %}
+{%                 endif %}
+      {{ redistribute_host }}
+{%             endif %}
+{%             if redistribute_var.connected.enabled is arista.avd.defined(true) %}
+{%                 set redistribute_conn = "redistribute connected" %}
+{%                 if redistribute_var.connected.route_map is arista.avd.defined %}
+{%                     set redistribute_conn = redistribute_conn ~ " route-map " ~ redistribute_var.connected.route_map %}
+{%                 endif %}
+      {{ redistribute_conn }}
+{%             endif %}
+{%             if redistribute_var.isis.enabled is arista.avd.defined(true) %}
+{%                 set redistribute_isis = "redistribute isis" %}
+{%                 if redistribute_var.isis.isis_level is arista.avd.defined %}
+{%                     set redistribute_isis = redistribute_isis ~ " " ~ redistribute_var.isis.isis_level %}
+{%                 endif %}
+{%                 if redistribute_var.isis.include_leaked is arista.avd.defined(true) %}
+{%                     set redistribute_isis = redistribute_isis ~ " include leaked" %}
+{%                 endif %}
+{%                 if redistribute_var.isis.route_map is arista.avd.defined %}
+{%                     set redistribute_isis = redistribute_isis ~ " route-map " ~ redistribute_var.isis.route_map %}
+{%                 elif redistribute_var.isis.rcf is arista.avd.defined %}
+{%                     set redistribute_isis = redistribute_isis ~ " rcf " ~ redistribute_var.isis.rcf %}
+{%                 endif %}
+      {{ redistribute_isis }}
+{%             endif %}
+{%             if redistribute_var.ospf.enabled is arista.avd.defined(true) %}
+{%                 set redistribute_ospf = "redistribute ospf" %}
+{%                 if redistribute_var.ospf.route_map is arista.avd.defined %}
+{%                     set redistribute_ospf = redistribute_ospf ~ " route-map " ~ redistribute_var.ospf.route_map %}
+{%                 elif redistribute_var.ospf.match_internal.enabled is arista.avd.defined(true) %}
+{%                     set redistribute_ospf = redistribute_ospf ~ " match internal" %}
+{%                     if redistribute_var.ospf.match_internal.route_map is arista.avd.defined %}
+{%                         set redistribute_ospf = redistribute_ospf ~ " route-map " ~ redistribute_var.ospf.match_internal.route_map %}
 {%                     endif %}
 {%                 endif %}
-{%                 if redistribute_route.include_leaked is arista.avd.defined and redistribute_route.source_protocol == "isis" %}
-{%                     set redistribute_route_cli = redistribute_route_cli ~ " include leaked" %}
+      {{ redistribute_ospf }}
+{%             endif %}
```

Comment:
here `redistribute_var.ospf.enabled` is not a `required key` and as per the logic here match_internal could not be configured if `redistribute_var.ospf.enabled is not true`
same at other places

Reply:
updated with or condition